eogrow/core/area/custom_grid.py

```diff
@@ -81,3 +81,46 @@ def _construct_file_path(self, *, prefix: str, suffix: str = "gpkg", **_: Any) -
         filename = filename.replace(" ", "_")
 
         return fs.path.combine(self.storage.get_cache_folder(), f"{filename}.{suffix}")
+
+
+class NewCustomGridAreaManager(BaseAreaManager):
+    """Area manager that works with a pre-defined grid of EOPatches"""
+
+    class Schema(BaseAreaManager.Schema):
+        grid_folder_key: str = Field("input_data", description="Which folder the grid file is in.")
+        grid_filename: Path = Field(
+            description=(
+                "A Geopackage with a collection of bounding boxes and attributes that will define EOPatches. In"
+                " case bounding boxes are in multiple CRS then each Geopackage layer should contain bounding boxes"
+                " from one CRS."
+            ),
+            regex=r"^.+\..+$",
+        )
+        name_column: str = Field(description="Name of the column containing EOPatch names.")
+
+    config: Schema
+
+    def _create_grid(self) -> Dict[CRS, gpd.GeoDataFrame]:
```

Comment:
Perhaps this functions name is misleading, since the grid already exists in this case. 

Maybe `_create_grid_dict` or `_create_grid_map` ? (if yes, fix also in the base class)

Reply:
I consider a `grid` to be an object of type `Dict[CRS, gpd.GeoDataFrame]`. And this method has to provide such a grid, given the input parameters. So here it does create such a grid, and it does so by loading a geopackage.

Would the name `_provide_grid` or `_prepare_grid` be more sensible in your opinion? The names kind of work for splitter-based classes and the batch one as well.

Reply:
you know what, it's actually fine, I had a wrong mapping of names in my head. my bad